### PR TITLE
Fixes #2714 - corrected IO#popen parameter on ruby 1.8.7

### PR DESCRIPTION
**Pull request by dmitri-d**

### CODE REVIEW

**Comment by ohadlevy** [comment]
@witlessbird all tests fail on 1.87 ;)

**Comment by dmitri-d** [comment]
Ugh. fixed.

File: test/runner_test.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+class RunnerTest < Test::Unit::TestCase
+  def setup
+    @runner = Proxy::Puppet::Runner.new({})
+  end
+
+  def test_popen
+    RUBY_VERSION > "1.8.7" ?
+      IO.expects(:popen).with(["sudo", "command", "anarg"]).returns(true) :
+      IO.expects(:popen).with("sudo command anarg").returns(true)
+    @runner.send(:popen, ["sudo", "command", "anarg"])
```
Comment (domcleal) [inline test/runner_test.rb]: How about testing a real command too, e.g. `echo`?

Comment (dmitri-d) [inline test/runner_test.rb]: I could, although it wouldn't matter - it's a mock...

Comment (domcleal) [inline test/runner_test.rb]: Sorry, I meant explicitly don't mock it and really do run a (harmless) command.

**Comment by dmitri-d** [comment]
@domcleal: how's this?

**Comment by domcleal** [comment]
Merged as 6139485, thanks @witlessbird!

**Pull request closed without merging**